src/Engine/ProtoCore/Core.cs

```diff
@@ -1510,17 +1510,20 @@ public void Cleanup()
 
         public void InitializeContextGlobals(Dictionary<string, object> context)
```

Comment:
InitializeContextGlobals is removed, AddContextData handles context injection generically for all types